src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -830,6 +826,12 @@ public string GenerateAuthHeader(AgentTaskPluginExecutionContext executionContex
                 {
                     additionalFetchSpecs.Add("+refs/heads/*:refs/remotes/origin/*");
                     additionalFetchSpecs.Add($"+{sourceBranch}:{GetRemoteRefName(sourceBranch)}");
+
+                    // If we are fetching by commit it may no longer be associated with the branch
+                    if (fetchByCommit && !string.IsNullOrEmpty(sourceVersion))
+                    {
+                        additionalFetchSpecs.Add($"{sourceVersion}");
```

Comment:
Could you clarify why the fetchDepth > 0 code path above this (line 820) uses the refspec `$"+{sourceVersion}:refs/remotes/origin/{sourceVersion}"` and this path uses just sourceVersion by itself? Also, this could be `additionalFetchSpecs.Add(sourceVersion)` without the interpolation if that is the intent.

Reply:
I updated the PR as we discussed.  This keeps current behavior with the addition of a fetch explicitly for commit when needed.